charts/trino/templates/deployment-coordinator.yaml

```diff
@@ -10,6 +10,16 @@ metadata:
     {{- tpl (toYaml .Values.coordinator.labels) . | nindent 4 }}
     {{- end }}
 spec:
+  {{- if .Values.coordinator.deployment.progressDeadlineSeconds }}
+  progressDeadlineSeconds: {{ .Values.coordinator.deployment.progressDeadlineSeconds }}
+  {{- end }}
+  {{- if or .Values.coordinator.deployment.revisionHistoryLimit (eq 0 .Values.coordinator.deployment.revisionHistoryLimit) }}
```

Comment:
It might be easier to read this condition if it checked if the value is not an empty string

Reply:
Sounds good.
Important thing to notice here is that, if the user sets `revisionHistoryLimit: 0` then they explicitly want to [clean up all history](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#clean-up-policy).
However in Helm, having `foo: 0`  is treated as if omitted.

We could either do:

1. `{{- if (ne "" .Values.coordinator.deployment.revisionHistoryLimit) }}` (maybe the easiest to read)
2. ` {{- if not (eq "" .Values.coordinator.deployment.revisionHistoryLimit) }}`

Reply:
Maybe it would be best to have a non-nil default? K8s docs say the default for `revisionHistoryLimit` is 10.

Reply:
Are these conditions needed at all, if the defaults are null?

Reply:
Fixed

Reply:
However @nineinchnick , `revisionHistoryLimit` is usually added as a `int` value in a `values.yaml`.

Comparing this value to a string in a helm template might be a bit verbose compared to `{{- if or .Values.coordinator.deployment.revisionHistoryLimit (eq 0 .Values.coordinator.deployment.revisionHistoryLimit) }}`.

Would you be ok keeping this as a check for `revisionHistoryLimit` ?

Reply:
What would happen if you remove the condition at all, and don't set the value, keeping the default null? Will Kubernetes apply its own default value?

Reply:
If you remove the condition at all then the `Spec` will be
```
revisionHistoryLimit:
```
I just tested submitting this to a cluster and it seems that k8s substitute with their default value.
So I think we are fine with this flow.

However, the only problem that could arise is if the user forgets to provide a value in the `values.yaml` file. Then it would break.  I feel this change is not backwards compatible with people using a holder tag (meaning older `values.yaml` file) of the chart.

My suggestion would be to have a condition.

Let me know your thoughts.

Reply:
I meant, if a user does not add `revisionHistoryLimit` to `values.yaml` file then this template will fail to render
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  ...
spec:
  progressDeadlineSeconds: {{ .Values.coordinator.deployment.progressDeadlineSeconds }}
  revisionHistoryLimit: {{ .Values.coordinator.deployment.revisionHistoryLimit }}
  {{- if .Values.coordinator.deployment.strategy }}
  strategy:
    {{- toYaml .Values.coordinator.deployment.strategy | nindent 4 }}
  {{- end }}
  ...
```

`Error: template: trino/templates/deployment-worker.yaml:18:19: executing "trino/templates/deployment-worker.yaml" at <.Values.coordinator.deployment.revisionHistoryLimit>: nil pointer evaluating interface {}.revisionHistoryLimit`

Having a condition that check if the value exists in the `values.yaml` file feels safer.

Reply:
What about using same defaults as Kubernetes does?

Reply:
Sounds good. I will modify